connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerTask.java

```diff
@@ -128,13 +125,10 @@ private void doClose() {
     private void doRun() {
         try {
             synchronized (this) {
-                if (stopping.get())
+                if (stopping)
                     return;
 
-                if (targetState.get() == TargetState.PAUSED)
-                    statusListener.onPause(id);
-                else
-                    statusListener.onStartup(id);
+                statusListener.onStartup(id);
```

Comment:
>  So it seems that, e.g., after a rebalance we will still create all the tasks and initialize them, but rely on shouldPause() checks to effectively put them to sleep. Is this the behavior we ultimately want or should paused tasks effectively not use any resources?

I deliberately opted for this in an attempt at keeping the flow straightforward, but it does have the drawback you mentioned. However I will make it so the task does not start at all until unpaused.